mpt_trie/examples/ethereum_trie.rs

```diff
@@ -36,34 +38,43 @@ struct StateTrieEntry {
     code_hash: H256,
 }
 
-fn main() {
+fn main() -> TrieOpResult<()> {
     let mut rng = StdRng::seed_from_u64(0);
 
-    let (account_entries, account_storage_tries): (Vec<_>, Vec<_>) = (0..NUM_ACCOUNTS_TO_GEN)
+    let generated: Result<Vec<_>, _> = (0..NUM_ACCOUNTS_TO_GEN)
         .map(|_| generate_fake_account_and_storage_trie(&mut rng))
-        .unzip();
+        .collect();
 
-    let _state_trie = StandardTrie::from_iter(
+    let (account_entries, account_storage_tries): (Vec<_>, Vec<_>) = match generated {
+        Ok(pairs) => pairs.into_iter().unzip(),
+        Err(e) => {
+            return Err(e);
+        }
+    };
+
+    let _state_trie = StandardTrie::try_from_iter(
         account_entries
             .into_iter()
             .map(|(k, acc)| (Nibbles::from_h256_be(k), acc.rlp_bytes().to_vec())),
-    );
+    )?;
 
     let _account_storage_tries: Vec<(AccountAddr, HashedPartialTrie)> = account_storage_tries;
 
```

Comment:
Huh... I wonder why I left this at the end?

Reply:
maybe you're planning to use it based on the TODO afterward